crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -0,0 +1,166 @@
+use editor::{Editor, EditorEvent};
+use gpui::{
+    div, list, AnyElement, AppContext, EventEmitter, FocusHandle, FocusableView,
+    InteractiveElement, IntoElement, ListState, ParentElement, Render, Styled, View, ViewContext,
+};
+use language::LanguageRegistry;
+use std::sync::Arc;
+use ui::prelude::*;
+use workspace::item::Item;
+use workspace::Workspace;
+
+use crate::OpenPreview;
+
+pub struct MarkdownPreviewView {
+    focus_handle: FocusHandle,
+    list_state: ListState,
+}
+
+impl MarkdownPreviewView {
+    pub fn register(workspace: &mut Workspace, _cx: &mut ViewContext<Workspace>) {
+        let languages = workspace.app_state().languages.clone();
+
+        workspace.register_action(move |workspace, _: &OpenPreview, cx| {
+            if workspace.has_active_modal(cx) {
+                cx.propagate();
+                return;
+            }
+            let languages = languages.clone();
+            let editor = workspace.active_item_as::<Editor>(cx).unwrap();
```

Comment:
This should not panic if someone runs this action outside of an editor. This can just be a noop, doesn't need a notification or anything.